### PR TITLE
fix: [PL-21464]: adding label singulars for resources

### PR DESCRIPTION
**Pull request by swarajcheguri-harness**
### Summary

- Adding singular labels for resources

**Before**

image

**After**
image

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/20-rbac/RouteDestinations.tsx [hunk 1]
```diff
@@ -75,6 +76,7 @@ RbacFactory.registerResourceTypeHandler(ResourceType.USER, {
 RbacFactory.registerResourceTypeHandler(ResourceType.USERGROUP, {
   icon: 'res-userGroups',
   label: 'common.userGroups',
+  labelSingular: 'common.userGroup',
```
Comment (reetika04) [inline src/modules/20-rbac/RouteDestinations.tsx:79]: Instead of labelSingular can we call it displayLabel? Why do we need this??

Comment (reetika04) [reply src/modules/20-rbac/RouteDestinations.tsx:79]: This makes it unmaintainable. Why not just use UserGroups (1) instead?

Comment (swarajcheguri-harness) [reply src/modules/20-rbac/RouteDestinations.tsx:79]: labelSingular is not just for displaying the label everywhere, its only for few occasions where we need singular label. And there are places where they are used  not only to display the number of resources but used in different scenario [like this one](https://harness.atlassian.net/browse/PL-27517)

**Review by jenil-harness — APPROVED** [review]

**Review by yc147 — DISMISSED** [review]

**Review by praneshg239 — DISMISSED** [review]

File: src/modules/10-common/strings/strings.en.yaml [hunk 2]
```diff
@@ -1096,5 +1096,17 @@ freezeListActiveBannerText: There are {{count}} ongoing deployment freeze window
 freezeListActiveBannerExpandedText: On the {{scope}} from <b>{{startTime}}</b> to <b>{{endTime}}</b>
 resourceIdentifier: Resource Identifier
 configureNav: Or configure your nav!
+singularLables:
+  authenticationSetting: 'Authentication Setting'
+  defaultSetting: Default Setting
+  governancePolicie: Governance Policie
```
Comment (chris-harness) [inline src/modules/10-common/strings/strings.en.yaml:1102]:
```suggestion
  governancePolicy: Governance Policy
```

File: src/modules/10-common/strings/strings.en.yaml [hunk 3]
```diff
@@ -1096,5 +1096,17 @@ freezeListActiveBannerText: There are {{count}} ongoing deployment freeze window
 freezeListActiveBannerExpandedText: On the {{scope}} from <b>{{startTime}}</b> to <b>{{endTime}}</b>
 resourceIdentifier: Resource Identifier
 configureNav: Or configure your nav!
+singularLables:
```
Comment (chris-harness) [inline src/modules/10-common/strings/strings.en.yaml:1099]:
```suggestion
singularLabels:
```

**Review by praneshg239 — APPROVED** [review]

**Review by harshilbits — APPROVED** [review]

**Review by jenil-harness — APPROVED** [review]

**Review by chris-harness — APPROVED** [review]

**Review by reetika04 — APPROVED** [review]

**Pull request merged**